internal/gengapic/genrest.go

```diff
@@ -629,24 +627,30 @@ func (g *generator) emptyUnaryRESTCall(servName string, m *descriptor.MethodDesc
 	// TODO(dovs): handle http headers
 	// TODO(dovs): handle deadlines
 	// TODO(dovs): handle call options
-	p("// The default (false) for the other options are fine.")
-	p("// Field names should be lowerCamel, not snake.")
-	p("m := protojson.MarshalOptions{AllowPartial: true, EmitUnpopulated: true, UseProtoNames: false}")
-	requestObject := "req"
-	if info.body != "" && info.body != "*" {
-		requestObject = "body"
-		p("body := req.Get%s()", snakeToCamel(info.body))
+	body := "nil"
+	if info.verb != "get" && info.verb != "delete" {
+		p("// The default (false) for the other options are fine.")
+		p("// Field names should be lowerCamel, not snake.")
+		p("m := protojson.MarshalOptions{AllowPartial: true, EmitUnpopulated: true, UseProtoNames: false}")
+		requestObject := "req"
+		if info.body != "" && info.body != "*" {
```

Comment:
I think this should be the main, outermost if statement instead of the http method checks themselves. GETs and DELETEs should not have body annotations, and RPCs using other HTTP methods may not as well.

In fact, we should have a generation error if there's a body annotation for GET or DELETE, because the query param logic will exclude those fields from the query string, so even if we silently omit a body, those fields will not get sent.

Reply:
To be more precise, what I mean is
```
if info.body != "" {
  assert(method!="get" && method!="delete")
  // do everything else
  ...
}
```

Reply:
> outermost if statement instead of the http method checks themselves

Yep, agreed.

Reply:
Done.